### PR TITLE
tests: Proto -> JSON serialization tests for roxctl non-central cmds

### PR DESCRIPTION
**Pull request by rhybrillou**
### Description

The PR is part of a PR split stack.
See #11772 description for more information
PR stack:
- #11772
  - #11900 
    - #11901 
      - #11902 
        - #11903 
          - #11904 
            - #11905 

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [x] added unit tests
- [x] modified existing tests

#### How I validated my change

Manual run of `go test -v` in each directory where a test was either created or modified

### CODE REVIEW

File: roxctl/common/report/report_test.go [hunk 1]
```diff
@@ -313,7 +315,170 @@ func TestJSONRemarks(t *testing.T) {
...
+			expectedError: errors.Wrap(capacityExhaustedErr, "could not write final newline for alerts"),
+		},
+	}
+
+	for name, tc := range errorCases {
+		t.Run(name, func(t *testing.T) {
+			writer := newTestWriter(tc.writeableLen)
+			err := JSONWithRemarks(writer, tc.alerts, tc.remarks)
+			require.Error(t, err)
+			assert.Equal(t, tc.expectedError.Error(), err.Error())
```
Comment (janisz) [inline roxctl/common/report/report_test.go:347]:
```suggestion
			assert.EqualError(t, err, tc.expectedError.Error())
```

File: roxctl/common/report/report_test.go [hunk 2]
```diff
@@ -313,7 +315,170 @@ func TestJSONRemarks(t *testing.T) {
...
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			writer := newTestWriter(tc.writeableLen)
+			err := JSON(writer, tc.alerts)
+			if tc.expectedError != nil {
+				require.Error(t, err)
+				assert.Equal(t, tc.expectedError.Error(), err.Error())
```
Comment (janisz) [inline roxctl/common/report/report_test.go:442]:
```suggestion
				assert.ErrorEqual(t, err, tc.expectedError.Error())
```

File: roxctl/common/report/report_test.go [hunk 3]
```diff
@@ -313,7 +315,170 @@ func TestJSONRemarks(t *testing.T) {
...
+}
+
+type testWriter struct {
+	writer strings.Builder
+	cap    int
+}
+
+func newTestWriter(capacity int) *testWriter {
+	if capacity == 0 {
+		capacity = 1024 * 1024 * 1024 * 2
```
Comment (janisz) [inline roxctl/common/report/report_test.go:458]: What is this number?

Comment (rhybrillou) [reply roxctl/common/report/report_test.go:460]: Added a documentation comment.

**Review by janisz — APPROVED** [review]

**Pull request merged**